azblob/zt_user_delegation_sas_test.go

```diff
@@ -1,26 +1,159 @@
 	}
 
-	// Create pipeline w/ OAuth to handle user delegation key obtaining
+	// Create pipeline to handle requests
 	p := NewPipeline(*ocred, PipelineOptions{})
 
+	// Prepare user delegation key
 	bsu = bsu.WithPipeline(p)
 	keyInfo := NewKeyInfo(currentTime, currentTime.Add(48*time.Hour))
-	cudk, err := bsu.GetUserDelegationCredential(ctx, keyInfo, nil, nil)
+	budk, err := bsu.GetUserDelegationCredential(ctx, keyInfo, nil, nil) //MUST have TokenCredential
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	return containerURL, containerName, blobURL, blobName, budk, currentTime, p
+}
+
+// Attempting to create User Delegation Key SAS with Incorrect Permissions, should return err
+func (s *aztestsSuite) TestUserDelegationSASIncorrectPermissions(c *chk.C) {
+	_, containerName, _, blobName, cudk, currentTime, _ := CreateUserDelegationKey(c)
+	// Prepare User Delegation SAS query for Container
+	_, err := BlobSASSignatureValues{
+		Protocol:      SASProtocolHTTPS,
+		StartTime:     currentTime,
+		ExpiryTime:    currentTime.Add(24 * time.Hour),
+		Permissions:   "rdq",
+		ContainerName: containerName,
+	}.NewSASQueryParameters(cudk)
+	c.Assert(err, chk.NotNil)
+
+	// Prepare User Delegation SAS query for Blob; returns err due to wrong permission
+	_, err = BlobSASSignatureValues{
+		Protocol:      SASProtocolHTTPS,
+		StartTime:     currentTime,
+		ExpiryTime:    currentTime.Add(24 * time.Hour),
+		Permissions:   "rdq",
+		ContainerName: containerName,
+		BlobName:      blobName,
+	}.NewSASQueryParameters(cudk)
+	c.Assert(err, chk.NotNil)
+}
+
+// Creates a container with no permissions, upload fails due to lack of permissions
+func (s *aztestsSuite) TestUserDelegationSASContainerNoPermissions(c *chk.C) {
+	containerURL, containerName, _, _, cudk, currentTime, p := CreateUserDelegationKey(c)
+	// Prepare User Delegation SAS query
+	cSAS, err := BlobSASSignatureValues{
+		Protocol:      SASProtocolHTTPS,
+		StartTime:     currentTime,
+		ExpiryTime:    currentTime.Add(24 * time.Hour),
+		ContainerName: containerName,
+	}.NewSASQueryParameters(cudk)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// Create anonymous pipeline
+	p = NewPipeline(NewAnonymousCredential(), PipelineOptions{})
+
+	// Create the container
+	_, err = containerURL.Create(ctx, Metadata{}, PublicAccessNone)
+	defer containerURL.Delete(ctx, ContainerAccessConditions{})
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// Craft a container URL w/ container UDK SAS
+	cURL := containerURL.URL()
+	cURL.RawQuery += cSAS.Encode()
+	cSASURL := NewContainerURL(cURL, p)
+
+	// Create blob; upload returns err due to lack of permissions
+	bblob := cSASURL.NewBlockBlobURL("test")
+	_, err = bblob.Upload(ctx, strings.NewReader("hello world!"), BlobHTTPHeaders{}, Metadata{}, BlobAccessConditions{}, DefaultAccessTier, nil, ClientProvidedKeyOptions{})
+	c.Assert(err, chk.NotNil)
+}
+
+// Creates a container with all permissions
+func (s *aztestsSuite) TestUserDelegationSASContainerAllPermissions(c *chk.C) {
```

Comment:
this one looks very similar to TestUserDelegationSASContainer, any reason why they aren't combined?